tests/test_phase_psf.py

```diff
@@ -0,0 +1,383 @@
+    atm2.append(ar2)
+    assert atm != atm2
+    # Swap the last screen in atm2 for the one that should match atm.
+    del atm2[-1]
+    atm2.append(atm[-1])
+    assert atm == atm2
+
+    # Test building from empty PhaseScreenList
+    atm3 = galsim.PhaseScreenList()
+    atm3.extend(atm2)
+    assert atm == atm2
+
+    # Test constructing from existing PhaseScreenList
+    atm4 = galsim.PhaseScreenList(atm3)
+    del atm4[-1]
+    assert atm != atm4
+    atm4.append(atm[-1])
+    assert atm == atm4
+
+    # Test swap
+    atm4[0], atm4[1] = atm4[1], atm4[0]
+    assert atm != atm4
+    atm4[0], atm4[1] = atm4[1], atm4[0]
+    assert atm == atm4
+
+    wf = atm.wavefront(aper)
+    wf2 = atm2.wavefront(aper)
+    wf3 = atm3.wavefront(aper)
+    wf4 = atm4.wavefront(aper)
+
+    np.testing.assert_array_equal(wf, wf2, "PhaseScreenLists are inconsistent")
+    np.testing.assert_array_equal(wf, wf3, "PhaseScreenLists are inconsistent")
+    np.testing.assert_array_equal(wf, wf4, "PhaseScreenLists are inconsistent")
+
+    # Check copy
+    import copy
+    # Shallow copy copies by reference.
+    atm5 = copy.copy(atm)
+    assert atm[0] == atm5[0]
+    assert atm[0] is atm5[0]
+    atm.advance()
+    assert atm[0] == atm5[0]
+    assert atm[0] is atm5[0]
+    # Deepcopy actually makes an indepedent object in memory.
+    atm5 = copy.deepcopy(atm)
+    assert atm[0] == atm5[0]
+    assert atm[0] is not atm5[0]
+    atm.advance()
+    assert atm[0] != atm5[0]
+
+    # Constructor should accept both list and indiv layers as arguments.
+    atm6 = galsim.PhaseScreenList(atm[0])
+    atm7 = galsim.PhaseScreenList([atm[0]])
+    assert atm6 == atm7
+    atm6 = galsim.PhaseScreenList(atm[0], atm[1])
+    atm7 = galsim.PhaseScreenList([atm[0], atm[1]])
+    atm8 = galsim.PhaseScreenList(atm[0:2])  # Slice returns PhaseScreenList, so this works too.
+    assert atm6 == atm7
+    assert atm6 == atm8
+
+    # Check some actual derived PSFs too, not just phase screens.  Use a small pupil_plane_size and
+    # relatively large pupil_plane_scale to speed up the unit test.
+    atm.advance_by(1.0)
+    do_pickle(atm)
+    atm.reset()
+    # Ignore warnings related to manually specifying pupil plane size and scale.
+    import warnings
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        kwargs = dict(exptime=0.06, diam=4.0, lam=500.0,
+                      pupil_plane_size=6.0, pupil_plane_scale=8.0/256)
+
+        psf = atm.makePSF(**kwargs)
+    do_pickle(psf)
+    do_pickle(psf, func=lambda x:x.drawImage(nx=20, ny=20, scale=0.1))
+
+    # Need to reset atm2 since both atm and atm2 reference the same layer objects (not copies).
+    # Not sure if this is a feature or a bug, but it's also how regular python lists work.
+    atm2.reset()
+    psf2 = atm2.makePSF(**kwargs)
+
+    atm3.reset()
+    psf3 = atm3.makePSF(**kwargs)
+
+    atm4.reset()
+    psf4 = atm4.makePSF(**kwargs)
```

Comment:
As written, these should be in a catch_warnings block too.  In python 3, the warnings get emitted every time, not just the first time, so these three makePSF calls emit warnings.  (In python 2, the first one is emitted, then gets ignored by the filter.  Then later passes over the warning don't get emitted.)

There are some in later tests too.  You can add some print statements in python 2.7 to see when it would happen -- "start ignore warnings", "end ignore warnings" here, and then "emitting warning" in the code where the warning is issued.  Or temporarily change the warning to a print statement.

However, this maybe indicates that the test should be altered?  Can we have a test that uses a "good" pupil plane scale?  Or is that just way too slow to fit into our intended <O(1) s time frame for unit tests?